src/lib/threadpool.c

```diff
@@ -150,15 +149,25 @@ static void worker(void *arg)
 	uv_mutex_lock(&mutex);
 	for (;;) {
 		/* `mutex` should always be locked at this point. */
-
-		/* Keep waiting while either no work is present or only slow I/O
-		   and we're at the threshold for that. */
-		while (xx__queue_empty(&wq)) {
+		/* Keep waiting while either no work is present */
+		while (xx__queue_empty(&wq) &&
+		       xx__queue_empty(&thread_queues[ta->idx])) {
+			//XXX printf("worker_e: idx=%u\n", ta->idx);
 			idle_threads += 1;
 			uv_cond_wait(&cond, &mutex);
 			idle_threads -= 1;
 		}
 
+		//XXX printf("worker: idx=%u\n", ta->idx);
+		/* XXX === Assinging work item affinity */
+		if (!xx__queue_empty(&thread_queues[ta->idx])) {
+		    q = xx__queue_head(&thread_queues[ta->idx]);
+		    //XXX printf("worker: q=%p idx=%u\n", q, ta->idx);
+		    xx__queue_remove(q);
+		    xx__queue_init(q);
+		    xx__queue_insert_head(&wq, q);
+		}
+
```

Comment:
Should the global `wq` be eliminated? It seems like it no longer serves a purpose now that each thread has its own queue

Reply:
I left this queue purposely for the sole reason that we will need to delay serving I/O requests coming to sqlite3 databases in each thread until a set of checkpoint operations is completed. 
In case if we choose this scenario such a queue could be a part of the mechanism.
I propose to leave it for now and remove it (just a matter of 1-2 hours) if needed.